### PR TITLE
Adding dragon API to build index without any thread.

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: common/src/writer.rs [hunk 1]
```diff
@@ -62,7 +62,7 @@ impl<W: TerminatingWrite> TerminatingWrite for CountingWriter<W> {
 pub struct AntiCallToken(());
 
 /// Trait used to indicate when no more write need to be done on a writer
-pub trait TerminatingWrite: Write + Send {
+pub trait TerminatingWrite: Write + Send + Sync {
```
Comment (PSeitz) [inline common/src/writer.rs:65]: Is this a requirement of `SimpleIndexWriter`?

Comment (fulmicoton) [reply common/src/writer.rs:65]: It is required to make `SimpleWriter: Send + Sync` which itself is required in Quickwit.

File: src/core/index_simple_writer.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+use crate::indexer::operation::AddOperation;
+use crate::indexer::segment_updater::save_metas;
+use crate::indexer::SegmentWriter;
+use crate::{Directory, Document, Index, IndexMeta, Opstamp, Segment};
+
+#[doc(hidden)]
+pub struct IndexSimpleWriter {
```
Comment (PSeitz) [inline src/core/index_simple_writer.rs]: `SingleSegmentIndexWriter` or `IndexSingleSegmentWriter` would be better I think

Comment (fulmicoton) [reply src/core/index_simple_writer.rs]: I renamed it SingleSegmentIndexWriter

File: src/core/index.rs [hunk 3]
```diff
@@ -877,4 +899,28 @@ mod tests {
         );
         Ok(())
     }
+
+    #[test]
+    fn test_simple_writer() -> crate::Result<()> {
```
Comment (shikhar) [inline src/core/index.rs]: `test_single_segment_writer`

**Review by PSeitz — APPROVED** [review]

**Pull request merged**